pkg/dataobj/metastore/metastore.go

```diff
@@ -16,7 +16,7 @@ type Metastore interface {
 	DataObjects(ctx context.Context, start, end time.Time, matchers ...*labels.Matcher) ([]string, error)
 
 	// StreamsIDs returns object store paths and stream IDs for all matching objects for the given matchers between [start,end]
-	StreamIDs(ctx context.Context, start, end time.Time, matchers ...*labels.Matcher) ([]string, [][]int64, error)
+	StreamIDs(ctx context.Context, start, end time.Time, matchers ...*labels.Matcher) ([]string, [][]int64, []int, error)
```

Comment:
Returning the number of sections per object in the `StreamIDs` function may not be the most suitable place, however, since only the physical planning is using that functionality right now, it is the most efficient way in terms of lookups, as well as code impact.